spec/controllers/members/dues_controller_spec.rb

```diff
@@ -17,6 +17,70 @@
     end
   end
 
+  describe "DELETE cancel" do
+
+    let!(:user) { login_as(:member, name: "Foo Bar", email: "someone@example.com") }
+
+    let(:params) do
+      {
+        user_id: user.id
+      }
+    end
+
+    subject(:cancel_dues) { delete :cancel, params }
+
+    context "when the user does not have a Stripe ID" do
+      it 'sets the flash and redirects to the manage dues page' do
+        expect(subject).to redirect_to members_user_dues_path(user)
+        expect(flash[:notice]).to include "You don't have an active membership dues subscription"
+      end
+    end
+
+    context "when the user already has a Stripe ID" do
+      before do
```

Comment:
@arbonap I just noticed there is another `before` block on line 69. I'm don't think this will break anything, but they should probably be combined into one block.